src/components/Icon/index.js

```diff
@@ -0,0 +1,49 @@
+import _ from 'underscore';
+import React, {memo} from 'react';
+import PropTypes from 'prop-types';
+import themeColors from '../../styles/themes/default';
+import variables from '../../styles/variables';
+import BRAND_ASSETS from './BRAND_ASSETS';
+import EXPENSICONS from './EXPENSICONS';
+
+const ICONS = _.extend(BRAND_ASSETS, EXPENSICONS);
+
+const propTypes = {
+    icon: PropTypes.oneOf(_.values(ICONS)).isRequired,
+    width: PropTypes.number,
+    height: PropTypes.number,
+    isEnabled: PropTypes.bool,
+};
+
+const defaultProps = {
+    width: variables.iconSizeNormal,
+    height: variables.iconSizeNormal,
+    isEnabled: false,
+};
+
+const Icon = (props) => {
+    const IconToRender = props.icon;
+    let fillColor = props.isEnabled ? themeColors.heading : themeColors.icon;
+
+    // Do not pass a fill color for brand assets
+    if (_.contains(_.values(BRAND_ASSETS), props.icon)) {
+        fillColor = undefined;
+    }
+
+    return (
+        <IconToRender
+            width={props.width}
+            height={props.height}
+            fill={fillColor}
+        />
+    );
+};
+
+Icon.propTypes = propTypes;
+Icon.defaultProps = defaultProps;
+
+export default memo(Icon);
```

Comment:
The purpose for memoizing the component was to boost performance. Icons will render often (and almost always with the same props), so this seems like the classic use-case for `React.memo`.

Reply:
That sounds reasonable to me. I just have a few questions...

>Icons will render often

I'm wondering if this will always be the case? Maybe some icons will not render often? And even if they did... is it expensive to render an SVG? Said another way, is there some specific performance issue that we are trying to prevent? Should we just wrap all the things in `React.memo()` in that case? Is there any reason why we might want to use `React.memo()` judiciously? Or is it no big deal?

Reply:
Here's an [article](https://dmitripavlutin.com/use-react-memo-wisely/) that provides some pretty good insight on when it is and is not a good idea to use `React.memo`. This diagram summarizes the heuristic pretty well:

![image](https://user-images.githubusercontent.com/47436092/103970140-64fa6e00-511c-11eb-88ef-186f91924e8a.png)

As far as I understand, it's a balancing act between the time it takes to compare the props for equality vs the time it takes for the component to render. I think if you had a component with no props (unlikely, but possible), then you would _always_ want to use `React.memo`, because the equality comparison is close to instantaneous.

The `Icon` component for sure meets the first three criteria in the diagram above, and it's a bit of a toss-up whether or not it meets the third. I have no idea how to measure and know for sure. Probably not a significant difference either way, but maybe @tgolen would have a good 6th sense as to whether or not this is a good time to use `React.memo`.

Reply:
Thanks for sharing this. I read the article and the impression I have goes something like...

If the time it takes to perform a `React.memo()` shallow comparison is _greater_ than the time it takes to render a component then we might actually be slowing things down by using `React.memo()`? 

The author does mention to [use profiling tools](https://reactjs.org/docs/optimizing-performance.html#profiling-components-with-the-chrome-performance-tab) to measure the benefits which sounds like a smart idea, but I might not pull them out in the first place unless something seemed slow.

So, I guess I'm sort of in the camp that we should use `memo`, `PureComponent`, etc when it is very clear why they are being used, there is some measurable benefit, and if we do not use them performance will degrade in an obvious way. 

That said, if there is no measurable benefit in the other direction and the component still does what it's supposed to maybe it doesn't matter? At least one person in [this SO post](https://stackoverflow.com/questions/53074551/when-should-you-not-use-react-memo) said you should "always" use it so I really don't know.

Reply:
Also whatever we decide we should turn this into a SO post bc this is a good chat.

Reply:
Yep, agreed. I want to wait to see what Tim G has to say about it.